xl-objects.go

```diff
@@ -150,6 +156,8 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64) (io.Read
 	if !IsValidObjectName(object) {
 		return nil, ObjectNameInvalid{Bucket: bucket, Object: object}
 	}
+	nsMutex.RLock(bucket, object)
+	defer nsMutex.RUnlock(bucket, object)
```

Comment:
> It won't panic, RLock() will succeed fine as it is a read lock. If it was a write-lock Lock() it would have blocked.
> The reason a second RLock() is held is so that in the go-routine we still have a Rlock because the function GetObject would have returned hence releasing the lock held at the beginning of the GetObject

Understood need to clear that up in the code, it can get confusing.